### PR TITLE
Fix for `_calculate_score_mask` in Attention Layer

### PR DESCRIPTION
**Pull request by Faisal-Alsrheed**
**The Issue:**

In the previous implementation, if `v_mask` was set to `None` and `use_causal_mask` was `True`, the method encountered a `ValueError`. 

This issue arose because the method attempted to perform operations with a `None` value, which is not compatible with the expected tensor operations.

**The Enhancement:**

The modified implementation introduces a more robust handling mechanism for such scenarios:

1. Handling `None` for `v_mask` with Causal Mask: When `use_causal_mask` is `True`, the method now correctly generates a causal mask regardless of whether `v_mask` is `None`. 

3. Handling `None` for `v_mask` without Causal Mask: If `use_causal_mask` is `False`, and `v_mask` is `None`, the method safely returns `None`.

### CODE REVIEW

File: keras/layers/attention/attention.py [hunk 1]
```diff
@@ -184,26 +184,27 @@ def _apply_scores(self, scores, value, scores_mask=None, training=False):
...
+            if v_mask is not None:
+                # Mask of shape [batch_size, 1, Tv].
+                v_mask = ops.expand_dims(v_mask, axis=-2)
+                return ops.logical_and(v_mask, causal_mask)
             return causal_mask
-        return ops.logical_and(v_mask, causal_mask)
+        else:
+            # If not using causal mask, return the value mask as is,
+            # or None if the value mask is not provided.
+            return v_mask if v_mask is not None else None
```
Comment (fchollet) [inline keras/layers/attention/attention.py:207]: This can just be `return v_mask`, no?

**Review by fchollet — COMMENTED** [review]
Thanks for the PR!

**Comment by Faisal-Alsrheed** [comment]
@fchollet  
Done and Thank you for your feedback :)

**Review by fchollet — APPROVED** [review]
LGTM, thank you!

**Comment by fchollet** [comment]
While the tests pass on torch + CPU, they fail with torch + GPU -- are you able to take a look at the issue? https://source.cloud.google.com/results/invocations/6b476ada-a30b-454a-aa2f-89e611027bac/targets/keras%2Fgithub%2Fubuntu%2Fgpu%2Ftorch%2Fcontinuous/log

**Comment by Faisal-Alsrheed** [comment]
> While the tests pass on torch + CPU, they fail with torch + GPU -- are you able to take a look at the issue? https://source.cloud.google.com/results/invocations/6b476ada-a30b-454a-aa2f-89e611027bac/targets/keras%2Fgithub%2Fubuntu%2Fgpu%2Ftorch%2Fcontinuous/log

@fchollet Yes, sure :)  
https://github.com/keras-team/keras/pull/19105
